src/XhrFileReader.js

```diff
@@ -304,8 +309,9 @@ class XhrFileReader extends MediaFileReader {
 }
 
 XhrFileReader._config = {
+  additionalXhrHeaders: [],
   avoidHeadRequests: false,
-  disallowedXhrHeaders: [],
+  disallowedXhrHeaders: {},
```

Comment:
This should still be an array. It's a list of header names.
`additionalXhrHeaders` on the other hand is a map between header names and values, so it should be an Object.